Sources/XCTest/XCAbstractTest.swift

```diff
@@ -26,6 +26,37 @@ public class XCTest {
         fatalError("Must be overridden by subclasses.")
     }
 
+    /// The `XCTestRun` subclass that will be instantiated when the test is run
+    /// to hold the test's results. Must be overridden by subclasses.
+    public var testRunClass: AnyClass {
+        fatalError("Must be overridden by subclasses.")
+    }
+
+    /// The test run object that executed the test, an instance of
+    /// testRunClass. If the test has not yet been run, this will be nil.
+    /// - Note: FIXME: This property is meant to be `internal(set)`. It is
+    ///   publicly settable for now due to a Swift compiler bug on Linux. To
+    ///   ensure compatibility of tests between swift-corelibs-xctest and Apple
+    ///   XCTest, you should not set this property. See
+    ///   https://bugs.swift.org/browse/SR-1129 for details.
+    public public(set) var testRun: XCTestRun? = nil
+
+    /// The method through which tests are executed. Must be overridden by
+    /// subclasses.
+    public func performTest(run: XCTestRun) {
+        fatalError("Must be overridden by subclasses.")
+    }
+
+    /// Creates an instance of the `testRunClass` and passes it as a parameter
+    /// to `performTest()`.
+    public func runTest() {
```

Comment:
Uh-oh, looks like we have a problem: https://github.com/modocache/XCTest.swift/blob/5b5a152150bacea9c373dcfb16a8384563cd4419/XCTest.swift#L18-L20

How does Swift determine which version to use here. I think we may need to file a Swift bug...